test/functional/Jsr292/src/com/ibm/j9/jsr292/indyn/IndyTest.java

```diff
@@ -373,30 +373,31 @@ public void test_boostrap_return_constant_MethodType() {
 		AssertJUnit.assertTrue(expected == mt);
 	}
 
-	// test that if resolved CallSite is null, the same error is rethrown
+	// Test to verify behaviour if a CallSite initially resolves to null
 	@Test(groups = { "level.extended" })
-	public void test_CallSiteNullErrorRethrown () {
+	public void test_CallSiteNullErrorRethrown() {
 		/* The bootstrap method associated with the indy call in test_CallSiteNullErrorRethrown
 		 * will return null the first time its called, and a valid CallSite for all repeat calls.
 		 */
 
-		// Java 8: NullPointerException is expected on the first run
-		// Java 11: BootstrapMethodError is expected on the first run
 		try {
 			com.ibm.j9.jsr292.indyn.GenIndyn.test_CallSiteNullErrorRethrown();
 			Assert.fail("BootstrapMethodError or NullPointerException should be thrown.");
-		} catch(BootstrapMethodError e) {
-			Assert.assertTrue(VersionCheck.major() >= 11);
+		} catch (BootstrapMethodError e) {
+			// Java 8 (with OJDK MHs): BoostrapMethodError is expected on the first run
+			// Java 11: BootstrapMethodError is expected on the first run
+			Assert.assertTrue((VersionCheck.major() >= 11) || isOpenJDKMHsEnabled());
```

Comment:
`isOpenJDKMHsEnabled` doesn't exist; it was just a suggestion for the helper's name; the helper will need to be created where the `openjdk.methodhandles` system property is checked.